### ISSUE TITLE
VPN connected to GoogleOneVPN(PixelVPN) cannot share.

### ISSUE THREAD

**Issue by Jialin0502** [Mygod/VPNHotspot#604]
# # Bug

I found that the upstream of Google VPN and the commonly used VPN software are different. After open wlan, Hot spot will be no network connection.  (used latest master version https://github.com/Mygod/VPNHotspot/commit/508967e0c14837feeed6d29527606bbd0b13b76d )

Using other VPNs can share normally.

Need to provide log? This VPN not very common,I  can together test this.

## Screenshot
<table>
  <tr>
<td><img src="https://github.com/user-attachments/assets/b669da4d-ee7d-42cd-908e-401130bcc4cc" width="400"></td>
    <td><img src="https://github.com/user-attachments/assets/11c26bef-843a-4d7c-b877-475d7635fc51" width="250"><img src="https://github.com/user-attachments/assets/98cc93a8-f430-4c05-856a-e29683592f43" width="250"></td>
</table>

**Comment by Mygod** [comment]
Logs please.

**Comment by Mygod** [comment]
Thanks! Can you try clicking "clean/reapply routing rules" and see if that fixes it? Otherwise please export another log, thanks again!

**Comment by GD2021** [comment]
Connect your phone to a Wi-Fi network and log in to the Google VPN. Once the app is open, enable the wireless tethering feature. The other phone, when connected to this Wi-Fi, will not have internet access.
[vpnhotspot_log.zip](https://github.com/user-attachments/files/17030246/vpnhotspot_log.zip)
![image](https://github.com/user-attachments/assets/7258507e-523d-470e-acda-f33fedb705f2)
![image](https://github.com/user-attachments/assets/7943cf3e-6312-4073-93dd-2b636a19e8b1)
![image](https://github.com/user-attachments/assets/ad14d4af-0249-4f33-872d-46a0aa58ab59)
![image](https://github.com/user-attachments/assets/88f9b78c-b488-464b-9aae-862ee6204d51)

**Comment by GD2021** [comment]
![image](https://github.com/user-attachments/assets/d83132d3-b3a8-487f-aba5-367651a369e8)

**Comment by Mygod** [comment]
Yes, but can you try clicking "clean/reapply routing rules" in the settings page of this app and see if that fixes it? Otherwise please export a new log. Thanks!

**Comment by GD2021** [comment]
[vpnhotspot-5041622348703627665.log](https://github.com/user-attachments/files/17069197/vpnhotspot-5041622348703627665.log)

Thank you for your concern, I've tried, but it still doesn't work. Thank you again for your concern.

**Comment by Mygod** [comment]
Indeed it looks like the response packets are not going through to your downstream devices. Not sure how to resolve this since I do not own a Pixel currently.

**Comment by httpsColon** [comment]
手机系统被厂商魔改导致的，我就成功过一次，v2ng的vpn模式网络共享给电脑成功了，之后再也无法复现

**Comment by yitong2333** [comment]
> Indeed it looks like the response packets are not going through to your downstream devices. Not sure how to resolve this since I do not own a Pixel currently.

I think you can open a proxy5 server by using everyproxy and then use VPNHotspot's hotspot to route the socks5 server, but VPNHotspot can't currently share the socks5 server via wifi hotspot

**Comment by OXERIS** [comment]
> 手机系统被厂商魔改导致的，我就成功过一次，v2ng的vpn模式网络共享给电脑成功了，之后就再也无法恢复了

我测试了一般的vpn和代理工具，比如clash，v2ray都可以，但是Google vpn不行

**Comment by Mygod** [comment]
How are you guys using Google VPN if it requires strong integrity? 🤔

**Comment by Mygod** [comment]
```
# tcpdump -nvvi any host 34.117.59.81
tcpdump: data link type LINUX_SLL2
tcpdump: listening on any, link-type LINUX_SLL2 (Linux cooked v2), snapshot length 262144 bytes
23:24:41.825657 wlan2 In  IP (tos 0x0, ttl 64, id 16294, offset 0, flags [DF], proto ICMP (1), length 84)
    192.168.136.148 > 34.117.59.81: ICMP echo request, id 3799, seq 1503, length 64
23:24:41.825887 ipsec7 Out IP (tos 0x0, ttl 63, id 16294, offset 0, flags [DF], proto ICMP (1), length 84)
    100.74.24.67 > 34.117.59.81: ICMP echo request, id 3799, seq 1503, length 64
23:24:41.840059 ipsec7 In  IP (tos 0x0, ttl 121, id 0, offset 0, flags [none], proto ICMP (1), length 84)
    34.117.59.81 > 100.74.24.67: ICMP echo reply, id 3799, seq 1503, length 64
23:24:42.917758 wlan2 In  IP (tos 0x0, ttl 64, id 16436, offset 0, flags [DF], proto ICMP (1), length 84)
    192.168.136.148 > 34.117.59.81: ICMP echo request, id 3799, seq 1504, length 64
23:24:42.917985 ipsec7 Out IP (tos 0x0, ttl 63, id 16436, offset 0, flags [DF], proto ICMP (1), length 84)
    100.74.24.67 > 34.117.59.81: ICMP echo request, id 3799, seq 1504, length 64
23:24:42.946428 ipsec7 In  IP (tos 0x0, ttl 121, id 0, offset 0, flags [none], proto ICMP (1), length 84)
    34.117.59.81 > 100.74.24.67: ICMP echo reply, id 3799, seq 1504, length 64
```

I've played around for a few hours and can't figure out why reply packets are not being routed correctly. Given that VPN by Google is a system app, it is likely that there is some advanced things that they are doing that are not accessible to a normal VPN app. I recommend simply using a different VPN like 1.1.1.1 for now.

**Comment by Mygod** [comment]
Are you talking about [this library](https://github.com/google/vpn-libraries)? It seems like it's used by the old Google One VPN instead.

Their apk is obfuscated and minimized and honestly I'm not sure why anyone wants to use this VPN instead of many other alternatives.

**Comment by stanleylhs** [comment]
> Are you talking about [this library](https://github.com/google/vpn-libraries)? It seems like it's used by the old Google One VPN instead.
> 
> Their apk is obfuscated and minimized and honestly I'm not sure why anyone wants to use this VPN instead of many other alternatives.

I justed installed this app hoping to share Pixel VPN, it's very useful in some countries, I would love to see this issue fixed thank you.

**Comment by Mygod** [comment]
This looks like an AOSP bug. A workaround has been identified and will be shipped in the next release.

**Comment by Mygod** [comment]
Recommendation for affected users: do not use Vpn by Google with this app for now. If you need VPN sharing to work reliably, use a WireGuard-based VPN instead. In practice it avoids this platform IKEv2/IPsec forwarding issue and usually performs better anyway, especially for throughput, latency, and network switching on mobile.

**Comment by Mygod** [comment]
[v2.19.2](https://github.com/Mygod/VPNHotspot/releases/tag/v2.19.2) released with this fix. Upstream AOSP issue: https://issuetracker.google.com/issues/501945027

**Comment by OXERIS** [comment]
> [v2.19.2](https://github.com/Mygod/VPNHotspot/releases/tag/v2.19.2) released with this fix. Upstream AOSP issue: https://issuetracker.google.com/issues/501945027

ohohoh,  that is awesome!!!!

But i have just relocked my pixel🥹, I can not use it now

But I am really interested in the machinism 

can you explain the machinism to me? what's the difference after you fix the issue? 
How do you fix it?
thank you
